chalice/app.py

```diff
@@ -0,0 +1,153 @@
+import json
+import traceback
+
+from functools import wraps
+from chalice import Chalice, NotFoundError, ChaliceViewError
+from chalicelib.config import logger
+from chalicelib.matrix_handler import LoomMatrixHandler
+from chalicelib.request_handler import RequestHandler, RequestStatus
+from chalicelib.sqs import SqsQueueHandler
+
+app = Chalice(app_name='matrix-service-api')
+app.debug = True
+
+# Replace handler here for supporting concatenation on other matrix formats
+mtx_handler = LoomMatrixHandler()
+
+
+def api_endpoint_decorator(api_endpoint_func):
```

Comment:
I know this was a request from @ttung, so I'm curious what he thinks. But, this just collapses every possible error to a 500 with a traceback. That doesn't seem like an improvement 😕

Reply:
@mckinsel I believe the way this is handled in the DSS is to have custom exception types that can be transformed into the right responses. This would not be hard to fix if we lift and modify this code.

https://github.com/HumanCellAtlas/data-store/blob/master/dss/error.py

Let's make a separate issue for this and submit it in a different PR.